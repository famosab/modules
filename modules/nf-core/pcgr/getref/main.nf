process PCGR_GETREF {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/51/517cc3a46129fc586191412f29246889124f4654640a53230aa203a2bcc1d7dc/data'
        : 'community.wave.seqera.io/library/coreutils_curl_gzip_tar:17a6ea9a6766c02a'}"

    input:
    tuple val(meta), val(bundleversion), val(genome)

    output:
    tuple val(meta), path("${bundleversion}"), emit: pcgrref
    path "versions.yml",                       emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def bundle = "pcgr_ref_data.${bundleversion}.${genome}.tgz"
    """
    curl -O https://insilico.hpc.uio.no/pcgr/${bundle}
    gzip -dc ${bundle} | tar xvf -

    mkdir ${bundleversion}
    mv data/ ${bundleversion}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        curl: \$(curl --version | head -1 | cut -d ' ' -f 2)
        tar: \$(tar --version | head -1 | cut -d ' ' -f 4)
        gzip: \$(gzip --version | head -1 | cut -d ' ' -f 2)
    END_VERSIONS
    """

    stub:
    """
    mkdir ${bundleversion}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        curl: \$(curl --version | head -1 | cut -d ' ' -f 2)
        tar: \$(tar --version | head -1 | cut -d ' ' -f 4)
        gzip: \$(gzip --version | head -1 | cut -d ' ' -f 2)
    END_VERSIONS
    """
}
