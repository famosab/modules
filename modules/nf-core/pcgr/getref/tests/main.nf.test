nextflow_process {

    name "Test Process PCGR_GETREF"
    script "../main.nf"
    process "PCGR_GETREF"

    tag "modules"
    tag "modules_nfcore"
    tag "pcgr"
    tag "pcgr/getref"

    test("human - 20250314 - grch38") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    '20250314',
                    'grch38'
                ]
                """
            }
        }

        then {
            assert process.success
            assertAll(
                { assert snapshot(
                    process.out,
                    path(process.out.versions[0]).yaml
                ).match() }
            )
        }

    }

    test("human - 20250314 - grch38 - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    '20250314',
                    'grch38'
                ]
                """
            }
        }

        then {
            assert process.success
            assertAll(
                { assert snapshot(
                    process.out,
                    path(process.out.versions[0]).yaml
                ).match() }
            )
        }

    }

}
