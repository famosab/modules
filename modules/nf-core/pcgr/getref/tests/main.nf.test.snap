{
    "human - 20250314 - grch38 - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,5e6cd06e2e1cb1e969fc5fdc57cbbb09"
                ],
                "pcgrref": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5e6cd06e2e1cb1e969fc5fdc57cbbb09"
                ]
            },
            {
                "PCGR_GETREF": {
                    "wget": "1.25.0"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-29T12:52:32.031963"
    }
}